src/main/java/ua/okwine/productexpirationdate/controller/ProductController.java

```diff
@@ -0,0 +1,136 @@
+package ua.okwine.productexpirationdate.controller;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.multipart.MultipartFile;
+import ua.okwine.productexpirationdate.entity.Product;
+import ua.okwine.productexpirationdate.entity.Provider;
+import ua.okwine.productexpirationdate.requestWrappers.ProductRequest;
+import ua.okwine.productexpirationdate.requestWrappers.ProductRequestWrapper;
+import ua.okwine.productexpirationdate.service.ProductService;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
+@Controller
+@RequestMapping("/products")
+public class ProductController {
+
+    private final ProductService productService;
+
+    @Autowired
+    public ProductController(ProductService productService) {
+        this.productService = productService;
+    }
+
+    @GetMapping("/main")
+    public String startPage() {
+        return "startpage";
+    }
+
+    @GetMapping("/newExpDateForm")
+    public String showFormForAdd(Model model) {
+        List<Provider> providers = productService.findAllProviders();
+        model.addAttribute("providers", providers);
+        model.addAttribute("product", new Product());
+
+        return "products/new-expiration-date-form";
+    }
+
+    @GetMapping("/list-products")
+    public String getAll(Model model) {
+        List<Product> productList = productService.findAll();
+
+        model.addAttribute("products", productList);
+
+        return "products/list-products";
+    }
+
+    @GetMapping("/productUploadForm")
+    public String providerUploadForm() {
+        return "products/product-upload-form";
+    }
+
+    @PostMapping("/importProductFromExcel")
+    public String importExcel(Model model, @RequestParam("file") MultipartFile file) {
+        String uploadDirectory = "src/main/resources/temp_files";
+        String fullPath = "src/main/resources/temp_files/" + file.getOriginalFilename();
+
+        Path fileNameAndPath = Paths.get(uploadDirectory, file.getOriginalFilename());
+        try {
+            Files.write(fileNameAndPath, file.getBytes());
+        } catch (IOException e) {
+            e.printStackTrace();
```

Comment:
you should consider using logging here instead of printStackTrace(). You can use @Slf4j annotation and see an example here https://github.com/FedosovMax/knight-todo/blob/master/src/main/java/com/knighttodo/knighttodo/rest/DayResource.java